src/sensors/mpu9250sensor.cpp

```diff
@@ -279,23 +279,23 @@ void MPU9250Sensor::startCalibration(int calibrationType) {
     // Blink calibrating led before user should rotate the sensor
     m_Logger.info("Gently rotate the device while it's gathering magnetometer data");
     ledManager.pattern(15, 300, 3000/310);
-    float *calibrationDataMag = (float*)malloc(calibrationSamples * 3 * sizeof(float));
+    MagnetoCalibration *magneto = new MagnetoCalibration();
     for (int i = 0; i < calibrationSamples; i++) {
         ledManager.on();
         int16_t mx,my,mz;
         imu.getMagnetometer(&mx, &my, &mz);
-        calibrationDataMag[i * 3 + 0] = my;
-        calibrationDataMag[i * 3 + 1] = mx;
-        calibrationDataMag[i * 3 + 2] = -mz;
-        Network::sendRawCalibrationData(calibrationDataMag, CALIBRATION_TYPE_EXTERNAL_MAG, 0);
+        magneto->sample(mx, my, mz);
+
+        float rawMagFloat[3] = { (float)mx, (float)my, (float)mz};
+        Network::sendRawCalibrationData(rawMagFloat, CALIBRATION_TYPE_EXTERNAL_MAG, 0);
         ledManager.off();
         delay(250);
     }
     m_Logger.debug("Calculating calibration data...");
 
     float M_BAinv[4][3];
-    CalculateCalibration(calibrationDataMag, calibrationSamples, M_BAinv);
-    free(calibrationDataMag);
+    magneto->current_calibration(M_BAinv);
+    delete magneto;
 
     m_Logger.debug("[INFO] Magnetometer calibration matrix:");
     m_Logger.debug("{");
```

Comment:
Below this (GitHub doesn't allow me to comment there), there's another code block that uses `getMotion9()`. Should that use `MagnetoCalibration` as well?

Reply:
referring to the gyro calibration? No, it shouldn't.

Magneto takes a bunch of data points that should all lie on a sphere, and figures out how to transform them so they are actually on a sphere. This requires a constant vector that can move around arbitrarily. For accelerometer calibration, this is the gravity vector, and we "move" it around by rotating the tracker. For magnetometer calibration, this is the magnetic field, and we "move" it around by rotating the tracker.

We don't have something similar for a gyroscope currently, so instead we only try to compute the bias -- the average of a bunch of values.

Aside: The goal of this PR is simply to update the uses of magneto that already exist, since I had to change the interface for my improvement to magneto. Therefore, I only replaced the uses of `CalculateCalibration`, which was the old interface to magneto.

Reply:
Okay, makes sense